shelly/src/main/java/au/com/jtribe/shelly/Call.java

```diff
@@ -0,0 +1,75 @@
+package au.com.jtribe.shelly;
+
+import android.Manifest;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Build;
+import android.support.annotation.RequiresPermission;
+
+/**
+ * Represents a phone call to a number using Intent.ACTION_CALL
+ */
+public class Call {
+
+  private String phoneNumber;
+
+  public Call() {
+  }
+
+  /**
+   * Adds a phone number to the Call object that will be called using the intent
+   *
+   * @param number Phone number that will be called
+   * @return Call object this method was called on for method chaining
+   */
+  @RequiresPermission(Manifest.permission.CALL_PHONE)
+  public Call number(String number) {
+    if (number == null) {
+      throw new IllegalArgumentException("number == null");
+    }
+    this.phoneNumber = number;
+    return this;
+  }
+
+  /**
+   * Creates an ACTION_CALL intent adding this object's fields into the intent
+   *
+   * @return The intent storing this objects data, can be used to open apps capable of placing
+   * phone calls
+   */
+  public Intent asIntent() {
+    Intent phoneIntent;
+    if (phoneNumber != null) {
+      phoneIntent = new Intent(Intent.ACTION_CALL);
+      phoneIntent.setData(Uri.parse("tel:" + phoneNumber));
+    } else {
+      throw new IllegalStateException("You must specify a number to call");
+    }
+
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+      phoneIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_DOCUMENT);
+    } else {
+      //noinspection deprecation
+      phoneIntent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_WHEN_TASK_RESET);
+    }
+
+    return phoneIntent;
+  }
+
+  /**
+   * Creates an ACTION_CALL chooser intent with this object's fields as data in the intent
+   * The chooser has no title
+   */
+  public Intent asChooserIntent() {
+    return Intent.createChooser(asIntent(), null);
+  }
+
+  /**
+   * Creates an ACTION_CALL chooser intent with this object's fields as data in the intent
+   *
+   * @param prompt The chooser's title is set by prompt
+   */
+  public Intent asChooserIntent(CharSequence prompt) {
```

Comment:
Annotate the method with `@CheckResult` and `@NonNull` and the parameter with `@NonNull`